File: ruby/rails/app/controllers/api/hello_controller.rb
module Api
  class HelloController < ApplicationController
    def index
      render json: {
        message: 'Hello from Ruby on Rails!',
        framework: 'Ruby on Rails',
        status: 'success',
        timestamp: Time.now.to_i
      }
    end

    def health
      render json: { status: 'healthy' }
    end
  end
end

File: java/micronaut/src/main/java/com/example/HelloController.java
package com.example;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;

import java.util.HashMap;
import java.util.Map;

@Controller("/api")
public class HelloController {

    @Get("/hello")
    @Produces(MediaType.APPLICATION_JSON)
    public Map<String, Object> hello() {
        Map<String, Object> response = new HashMap<>();
        response.put("message", "Hello from Micronaut!");
        response.put("framework", "Micronaut");
        response.put("status", "success");
        response.put("timestamp", System.currentTimeMillis());
        return response;
    }

    @Get("/health")
    @Produces(MediaType.APPLICATION_JSON)
    public Map<String, String> health() {
        Map<String, String> response = new HashMap<>();
        response.put("status", "healthy");
        return response;
    }
}
